docs/howto/operate/new-cluster/new-cluster.md

```diff
@@ -72,12 +72,12 @@ See the [variables file](https://github.com/2i2c-org/infrastructure/blob/HEAD/te
 Names for Azure container registries and storage accounts **must** conform to the following guidelines:
 
 - alphanumeric strings between 5 and 50 characters for [container registries](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftcontainerregistry), e.g., `myContainerRegistry007`
-- alphanumeric strings between 2 and 24 characters for [storage accounts](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftstorage), e.g., `myStorageAccount314`
+- lowercase letters and numbers strings between 2 and 24 characters for [storage accounts](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftstorage), e.g., `mystorageaccount314`
 
 We explictly recommend the following conventions using `lowerCamelCase`:
 
-- `{CLUSTER_NAME}HubRegistry` for container registries
-- `{CLUSTER_NAME}HubStorage` for storage accounts
+- `{CLUSTER_NAME}hubregistry` for container registries
```

Comment:
```suggestion
- `{CLUSTER_NAME}HubRegistry` for container registries
```

So we _can_ use upper case in container registry names, the lower case restriction only applies to storage accounts. So the question is, what standard do we want to go with?

Reply:
Let's just go with CamelCase since that's what we've had documented so far, and add a note that in some cases technical limitations might prevent this from happening, in which case it's OK to just use all-lowercase.

Reply:
I have no strong preference either way. I think the important thing to document is that _a failure will occur if you try to create a storage account whose name is not 100% lowercase_.

Reply:
> Let's just go with CamelCase since that's what we've had documented so far, and add a note that in some cases technical limitations might prevent this from happening, in which case it's OK to just use all-lowercase.

We have documented CamelCase indeed, but all existing Azure clusters use lowercase. My preference is use lower case for both, for the sake of consistency and keeping it simple. Otherwise, we would end up with all sorts of naming schemas: "-", CamelCase and lower case:

```
subscription_id                = "my-awesome-subscription-id"
resourcegroup_name             = "my-awesome-resource-group"
global_container_registry_name = "myAwesomeHubRegistry"
global_storage_account_name    = "myawesomestorageaccount"
ssh_pub_key                    = "ssh-rsa my-public-ssh-key"
```

> I think the important thing to document is that a failure will occur if you try to create a storage account whose name is not 100% lowercase.

:+1:

Reply:
ah - in that case I think our decision criteria should be:

- A. If we expect to be *required* to use `lower-case` only occasionally, then we should use `CamelCase` and on rare occasions use `lower-case`
- B. If we instead expect to be required to use `lower-case` fairly often, and as long as we can *always* use `lower-case` for naming, then I think we should just standardize on `lower-case`

So which seems more likely?

Reply:
I think A is most likely. For example, I know `CamelCase` is allowed for the Container Registry.

> We have documented CamelCase indeed, but all existing Azure clusters use lowercase. My preference is use lower case for both, for the sake of consistency and keeping it simple. Otherwise, we would end up with all sorts of naming schemas: "-", CamelCase and lower case:

@GeorgianaElena I also the Azure terraform config could use a lot of cleaning up to bring it in-line with the GCP config, we can standardise during that cleanup process.

Reply:
Even if we are in scenario A, I do concur with @GeorgianaElena about the fact that mixing "schemas" could be potentially confusing...

Reply:
Sorry, I meant the _requirement to use lowercase only occasionally_ part of the statement is most likely.

Reply:
> requirement to use lowercase only occasionally

Still, I think it might introduce a waste of time and confusion but...

> I also the Azure terraform config could use a lot of cleaning up to bring it in-line with the GCP config, we can standardise during that cleanup process.

if we are really going to standardize later I guess I would be OK with a temporary scenario with "mixed schemas".

Reply:
> Still, I think it might introduce a waste of time and confusion but...

Yikes, I'm sorry, I'm still not doing well at communicating this.

The occasional requirement for lowercase in some resource names comes from Azure. So I was saying that if we wanted to standard on CamelCase, we would only occasionally run into this error. I still have no preference of what standard we adopt. I think the important thing to document is that whatever standard we adopt, Azure's own requirements will probably break it at some point. Because Azure is annoying.

Reply:
> I think the important thing to document is that whatever standard we adopt, Azure's own requirements will probably break it at some point. Because Azure is annoying.

💯